### PR TITLE
Bump to latest Shipwright

### PR DESCRIPTION
**Pull request by qu1queee**
This is required so that someone deploying Shipwright/Build from `master` branch can have compatibility with the resources generated from Shipwright/CLI.

At the moment CLI points to the Shipwright/Build version with the old group `build.dev`, while the latest Shipwright/Build uses `shipwright.io`

I struggle a lot doing this bump due to an issue with the `kustomize` dependency. The `go.mod` needs to ping a dependency due to some indirect dependencies in knative side that comes when bumping the shipwright/build module.

### CODE REVIEW

**Comment by qu1queee** [comment]
@alicerum pls take a look, we need this for cli to be compatible with latest changes upstream in build.

File: go.mod [hunk 1]
```diff
@@ -3,9 +3,9 @@ module github.com/shipwright-io/cli
 go 1.15
 
 require (
-	github.com/onsi/gomega v1.10.1
+	github.com/onsi/gomega v1.10.3
 	github.com/pkg/errors v0.9.1
-	github.com/shipwright-io/build v0.3.0
+	github.com/shipwright-io/build v0.3.1-0.20210330182238-23d2672f2f61
```
Comment (alicerum) [inline go.mod:8]: For `CLI`, do we want to follow nightly in our `main` branch and only depend on releases when we release `CLI`?
@adambkaplan @otaviof @qu1queee

**Comment by alicerum** [comment]
/lgtm

i'm not sure whether we want to wait for `build` release though or not

**Comment by adambkaplan** [comment]
@alicerum you may need to rebase this PR to get the OWNERS file updated.

**Comment by adambkaplan** [comment]
https://github.com/openshift/release/pull/17632 should fix the merge issue. Note this needs the "approve" label.

**Review by adambkaplan — APPROVED** [review]
/approve

**Comment by adambkaplan** [comment]
Going to "green button" this, need to fix one more thing before we get the tide automation working.

**Pull request merged**